crates/viewer/re_view_dataframe/src/display_record_batch.rs

```diff
@@ -151,6 +151,46 @@ impl ComponentData {
                 data
             };
 
+            // TODO(ab): somehow address this hack.
+            //
+            // Background: the `row_id` passed to `ui_raw` is only ever used as a cache key for
+            // images, and we must provide one for images to be displayed. Since in general we don't
+            // have a row id available (e.g. for an arbitrary table entry, or for a dataframe query
+            // result), we just hash the data to get a deterministic row id.
+            let row_id = (component_name.as_str() == "rerun.components.Blob")
```

Comment:
At this point we should no longer be calling it `row_id`. It should be something like `blob_hash: Hash64` instead.

Reply:
I used `row_id` to be consistent with the `DataUi` api, but I agree we should change it all:
- https://github.com/rerun-io/rerun/issues/9514

Reply:
Let's do that issue first then. This is, as far as I know, the only place where we create a `RowId` that isn't actually a row id. It's a lie. Using it to look up a row will always fail. Calling `.nanos_since_epoch()` on it will return nonsense. Let's not break our types intentionally.

Reply:
Disagree. The follow-up PR routes row id (if any) all the way to this place. It makes sense to keep this consistent. Also disagree doing the other issue first. It _should_ be easy, but maybe it's not, and this is high priority demo material.

Reply:
Passing around a `RowId` that is not a `RowId` is worrying tech debt, and a foot-gun lying around waiting to go off. If we need to move fast for the demo: fine; but then make sure the follow-up issue is added to next release milestone so we don't just accumulate tech debt.

Reply:
It was on my short term todo. I've added it to the milestone.